tests/test_processors.go

```diff
@@ -0,0 +1,148 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package tests
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+
+	"github.com/elastic/apm-server/decoder"
+	"github.com/elastic/apm-server/processor"
+	"github.com/elastic/apm-server/processor/stream"
+	"github.com/elastic/apm-server/publish"
+	"github.com/elastic/apm-server/tests/loader"
+	"github.com/elastic/apm-server/transform"
+	"github.com/elastic/beats/libbeat/beat"
+)
+
+type V1TestProcessor struct {
+	processor.Processor
+}
+
+func (p *V1TestProcessor) LoadPayload(path string) (interface{}, error) {
+	return loader.LoadData(path)
+}
+
+func (p *V1TestProcessor) Decode(input interface{}) error {
+	_, _, err := p.Processor.Decode(input.(map[string]interface{}))
+	return err
+}
+
+func (p *V1TestProcessor) Validate(input interface{}) error {
+	return p.Processor.Validate(input.(map[string]interface{}))
+}
+
+func (p *V1TestProcessor) Process(buf []byte) ([]beat.Event, error) {
+	var pl map[string]interface{}
+	err := json.Unmarshal(buf, &pl)
+	if err != nil {
+		return nil, err
+	}
+
+	err = p.Processor.Validate(pl)
```

Comment:
what's the advantage? For `Decode`, we need the returned `metadata` and `transformables`. `p.Decode` only returns an `error`

Reply:
Why can't you return that from decode? I just tried to avoid having the  `Processor.Validate` and `Processor.Decode` duplicated. But it is not a big deal.

Reply:
I'm not sure what you mean. Do you want me to call `p.Decode` instead of `p.Processor.Decode` and `p.Validate` instead of `p.Processor.Validate`?

Reply:
yes, use `p.Decode` instead of `p.Processor.Decode`, same for validation - as you are duplicating the logic from the processor here (which I thought you wanted to avoid with this implementation). But as I said, it is a minor detail, so if you don't agree, it's fine with me to leave as is.